### DISCUSSION TITLE
Consider including in core some methods from solidus_support

### DISCUSSION THREAD

**Discussion by kennyadsl** [solidusio/solidus#4597] [New Features or Ideas]
We are using [here](https://github.com/solidusio/solidus/blob/3802d57dbb86b111fa52bf9aceb9c475aa51ebd6/backend/app/views/spree/admin/products/edit.html.erb#L7) some code that is already wrapped into useful [methods of solidus_core gem](https://github.com/solidusio/solidus_support/blob/0dd30545a926fda128fa835957299c186a70fe46/lib/solidus_support.rb#L44-L54). 

[`solidus_support`](https://github.com/solidusio/solidus_support) is a gem that is meant to be included by other extension to provide useful methods and uniform behaviors among extensions.

I think it could be a good idea to include those methods in core and remove them from ~`solidus_core`~ `solidus_support` since they could be useful in core as well. 

Another option could be adding `solidus_support` gem in core - IMHO the cons of this is that we also add to core things that will not directly be used in core and should stay into extensions only, but we probably already have something like that in core.

This issue needs a deeper look at what's the best path for this.

Any feedback is welcome!

**Comment by nvh0412** [comment]
IMO, we can come up with the first one, including those methods in core but just a bit confuse with this one
>remove them from solidus_core since they could be useful in core as well

As for option 2, we will have to maintain `solidus_support` gem, these methods that we used inside solidus-core as well if we update our methods in `solidus_support`.

**Comment by kennyadsl** [comment]
> but just a bit confuse with this one

Sorry I mean `solidus_support` (and fixed the issue text)

> As for option 2, we will have to maintain solidus_support gem, these methods that we used inside solidus-core as well if we update our methods in solidus_support.

Good one  👍
